File: Golang/Chapter 3/Deploying our wallet.go
package main

import (
	"context"
	"crypto/ed25519"
	"crypto/hmac"
	"crypto/sha512"
	"encoding/base64"
	"github.com/xssnick/tonutils-go/address"
	"github.com/xssnick/tonutils-go/liteclient"
	"github.com/xssnick/tonutils-go/tl"
	"github.com/xssnick/tonutils-go/tlb"
	"github.com/xssnick/tonutils-go/ton"
	"github.com/xssnick/tonutils-go/ton/wallet"
	"github.com/xssnick/tonutils-go/tvm/cell"
	"golang.org/x/crypto/pbkdf2"
	"log"
	"strings"
	"time"
)

func main() {
	// mnemonic := strings.Split("put your mnemonic", " ") // get our mnemonic as array
	mnemonic := wallet.NewSeed() // get new mnemonic

	// The following three lines will extract the private key using the mnemonic phrase. We will not go into cryptographic details. It has all been implemented in the tonutils-go library, but it immediately returns the finished object of the wallet with the address and ready methods. So we’ll have to write the lines to get the key separately. Goland IDE will automatically import all required libraries (crypto, pbkdf2 and others).
	mac := hmac.New(sha512.New, []byte(strings.Join(mnemonic, " ")))
	hash := mac.Sum(nil)
	k := pbkdf2.Key(hash, []byte("TON default seed"), 100000, 32, sha512.New) // In TON libraries "TON default seed" is used as salt when getting keys
	// 32 is a key len

	privateKey := ed25519.NewKeyFromSeed(k)              // get private key
	publicKey := privateKey.Public().(ed25519.PublicKey) // get public key from private key
	log.Println(publicKey)                               // print publicKey so that at this stage the compiler does not complain that we do not use our variable
	log.Println(mnemonic)                                // if we want, we can print our mnemonic

	var subWallet uint64 = 698983191

	base64BOC := "te6ccgEBCAEAhgABFP8A9KQT9LzyyAsBAgEgAgMCAUgEBQCW8oMI1xgg0x/TH9MfAvgju/Jj7UTQ0x/TH9P/0VEyuvKhUUS68qIE+QFUEFX5EPKj+ACTINdKltMH1AL7AOgwAaTIyx/LH8v/ye1UAATQMAIBSAYHABe7Oc7UTQ0z8x1wv/gAEbjJftRNDXCx+A==" // save our base64 encoded output from compiler to variable
	codeCellBytes, _ := base64.StdEncoding.DecodeString(base64BOC)                                                                                                                                                      // decode base64 in order to get byte array
	codeCell, err := cell.FromBOC(codeCellBytes)                                                                                                                                                                        // get cell with code from byte array
	if err != nil {                                                                                                                                                                                                     // check if there are any error
		panic(err)
	}

	log.Println("Hash:", base64.StdEncoding.EncodeToString(codeCell.Hash())) // get the hash of our cell, encode it to base64 because it has []byte type and output to the terminal

	dataCell := cell.BeginCell().
		MustStoreUInt(0, 32).           // Seqno
		MustStoreUInt(698983191, 32).   // Subwallet ID
		MustStoreSlice(publicKey, 256). // Public Key
		EndCell()

	stateInit := cell.BeginCell().
		MustStoreBoolBit(false). // No split_depth
		MustStoreBoolBit(false). // No special
		MustStoreBoolBit(true).  // We have code
		MustStoreRef(codeCell).
		MustStoreBoolBit(true). // We have data
		MustStoreRef(dataCell).
		MustStoreBoolBit(false). // No library
		EndCell()

	contractAddress := address.NewAddress(0, 0, stateInit.Hash()) // get the hash of stateInit to get the address of our smart contract in workchain with ID 0
	log.Println("Contract address:", contractAddress.String())    // Output contract address to console

	internalMessageBody := cell.BeginCell().
		MustStoreUInt(0, 32).
		MustStoreStringSnake("Hello, TON!").
		EndCell()

	internalMessage := cell.BeginCell().
		MustStoreUInt(0x10, 6). // no bounce
		MustStoreAddr(address.MustParseAddr("put your first wallet address from were you sent 0.1 TON")).
		MustStoreBigCoins(tlb.MustFromTON("0.03").NanoTON()).
		MustStoreUInt(1, 1+4+4+64+32+1+1). // We store 1 that means we have body as a reference
		MustStoreRef(internalMessageBody).
		EndCell()

	// transaction for our wallet
	toSign := cell.BeginCell().
		MustStoreUInt(subWallet, 32).
		MustStoreUInt(uint64(time.Now().UTC().Unix()+60), 32).
		MustStoreUInt(0, 32). // We put seqno = 0, because after deploying wallet will store 0 as seqno
		MustStoreUInt(3, 8).
		MustStoreRef(internalMessage)

	signature := ed25519.Sign(privateKey, toSign.EndCell().Hash())
	body := cell.BeginCell().
		MustStoreSlice(signature, 512).
		MustStoreBuilder(toSign).
		EndCell()

	externalMessage := cell.BeginCell().
		MustStoreUInt(0b10, 2). // indicate that it is an incoming external transaction
		MustStoreUInt(0, 2).    // src -> addr_none
		MustStoreAddr(contractAddress).
		MustStoreCoins(0).       // Import fee
		MustStoreBoolBit(true).  // We have State Init
		MustStoreBoolBit(true).  // We store State Init as a reference
		MustStoreRef(stateInit). // Store State Init as a reference
		MustStoreBoolBit(true).  // We store Message Body as a reference
		MustStoreRef(body).      // Store Message Body as a reference
		EndCell()

	connection := liteclient.NewConnectionPool()
	configUrl := "https://ton-blockchain.github.io/global.config.json"
	err = connection.AddConnectionsFromConfigUrl(context.Background(), configUrl)
	if err != nil {
		panic(err)
	}
	client := ton.NewAPIClient(connection)

	var resp tl.Serializable
	err = client.Client().QueryLiteserver(context.Background(), ton.SendMessage{Body: externalMessage.ToBOCWithFlags(false)}, &resp)
	if err != nil {
		log.Fatalln(err.Error())
		return
	}
}

File: JavaScript - TypeScript/src/Chapter 3/Deploying our wallet.ts
import { compileFunc } from "@ton-community/func-js";
import { Address, Cell, beginCell, toNano } from "@ton/core";
import { mnemonicNew, mnemonicToWalletKey, sign } from "@ton/crypto";
import { TonClient } from "@ton/ton";
import fs from 'fs'; // we use fs for reading content of files

async function main() {
    // const mnemonicArray = 'put your mnemonic'.split(' ') // get our mnemonic as array
    const mnemonicArray = await mnemonicNew(24); // 24 is the number of words in a seed phrase
    const keyPair = await mnemonicToWalletKey(mnemonicArray); // extract private and public keys from mnemonic
    console.log(mnemonicArray) // if we want, we can print our mnemonic

    const subWallet = 698983191;

    const result = await compileFunc({
        targets: ['wallet_v3.fc'], // targets of your project
        sources: {
            "stdlib.fc": fs.readFileSync('./src/stdlib.fc', { encoding: 'utf-8' }),
            "wallet_v3.fc": fs.readFileSync('./src/wallet_v3.fc', { encoding: 'utf-8' }),
        }
    });

    if (result.status === 'error') {
        console.error(result.message)
        return;
    }

    const codeCell = Cell.fromBoc(Buffer.from(result.codeBoc, "base64"))[0]; // get buffer from base64 encoded BOC and get cell from this buffer

    // now we have base64 encoded BOC with compiled code in result.codeBoc
    console.log('Code BOC: ' + result.codeBoc);
    console.log('\nHash: ' + codeCell.hash().toString('base64')); // get the hash of cell and convert in to base64 encoded string. We will need it further

    const dataCell = beginCell().
        storeUint(0, 32). // Seqno
        storeUint(698983191, 32). // Subwallet ID
        storeBuffer(keyPair.publicKey). // Public Key
        endCell();

    const stateInit = beginCell().
        storeBit(0). // No split_depth
        storeBit(0). // No special
        storeBit(1). // We have code
        storeRef(codeCell).
        storeBit(1). // We have data
        storeRef(dataCell).
        storeBit(0). // No library
        endCell();

    const contractAddress = new Address(0, stateInit.hash()); // get the hash of stateInit to get the address of our smart contract in workchain with ID 0
    console.log(`Contract address: ${contractAddress.toString()}`); // Output contract address to console

    const internalMessageBody = beginCell().
    storeUint(0, 32).
    storeStringTail("Hello, TON!").
    endCell();

    const internalMessage = beginCell().
    storeUint(0x10, 6). // no bounce
    storeAddress(Address.parse("put your first wallet address from were you sent 0.1 TON")).
    storeCoins(toNano("0.03")).
    storeUint(1, 1 + 4 + 4 + 64 + 32 + 1 + 1). // We store 1 that means we have body as a reference
    storeRef(internalMessageBody).
    endCell();

    // transaction for our wallet
    const toSign = beginCell().
    storeUint(subWallet, 32).
    storeUint(Math.floor(Date.now() / 1e3) + 60, 32).
    storeUint(0, 32). // We put seqno = 0, because after deploying wallet will store 0 as seqno
    storeUint(3, 8).
    storeRef(internalMessage);

    const signature = sign(toSign.endCell().hash(), keyPair.secretKey);
    const body = beginCell().
    storeBuffer(signature).
    storeBuilder(toSign).
    endCell();

    const externalMessage = beginCell().
        storeUint(0b10, 2). // indicate that it is an incoming external transaction
        storeUint(0, 2). // src -> addr_none
        storeAddress(contractAddress).
        storeCoins(0). // Import fee
        storeBit(1). // We have State Init
        storeBit(1). // We store State Init as a reference
        storeRef(stateInit). // Store State Init as a reference
        storeBit(1). // We store Message Body as a reference
        storeRef(body). // Store Message Body as a reference
        endCell();

    const client = new TonClient({
        endpoint: "https://toncenter.com/api/v2/jsonRPC",
        apiKey: "put your api key" // you can get an api key from @tonapibot bot in Telegram
    });
        
    client.sendFile(externalMessage.toBoc());
}

main().finally(() => console.log("Exiting..."));